frontend/src/pages/home/AlternatingBackgroundColorSectionList.tsx

```diff
@@ -2,45 +2,105 @@ import React from 'react';
+      sx={{
+        display: 'flex',
+        justifyContent: 'center',
+        padding: verticalPadding,
+        [theme.breakpoints.down('md')]: {
+          padding: `${verticalPadding} 16px ${verticalPadding} 16px`,
+        },
+        color: color,
+        background: background,
+      }}
+    >
+      {child}
+    </Grid>
+  );
+};
+
 const AlternatingBackgroundColorSectionList = ({
   children,
   secondaryBackground = '#1282B2',
   secondaryColor = 'white',
+  verticalPadding = '48px',
+  firstElemVerticalPadding = '32px',
+  lastItemIsPrimary,
 }: {
   children: React.ReactNode[];
   secondaryBackground?: string;
   secondaryColor?: string;
+  // Allow the first item to have a different padding than the rest of the
+  // items.
+  firstElemVerticalPadding?: string;
+  // Used to wrap every item.
+  verticalPadding?: string;
+  // If true the last item will use the primary color regardless of its
+  // position in the list.
+  lastItemIsPrimary?: boolean;
 }) => {
   const theme = useTheme();
 
+  const isLastItem = (idx: number) => {
+    return idx === children.length - 1;
+  };
+
   return (
-    <div
-      style={{
-        width: '100%',
-      }}
-    >
-      <Grid container>
-        {children.map((child: React.ReactNode, i: number) => (
-          <Grid
+    <Grid container width="100%">
+      {children.map((child: React.ReactNode, i: number) =>
+        isLastItem(i) ? (
+          <SectionGridItem
+            key={i}
+            child={child}
+            verticalPadding={verticalPadding}
+            color={
+              lastItemIsPrimary
+                ? theme.palette.text.primary
+                : i % 2 == 0
+                ? secondaryColor
+                : theme.palette.text.primary
+            }
+            background={
+              lastItemIsPrimary
+                ? theme.palette.background.primary
+                : i % 2 == 0
+                ? secondaryBackground
+                : theme.palette.background.primary
+            }
+          />
+        ) : (
+          <SectionGridItem
             key={i}
-            item
-            xs={12}
-            sx={{
-              display: 'flex',
-              justifyContent: 'center',
-              padding: '32px',
-              [theme.breakpoints.down('md')]: {
-                padding: '32px 16px 32px 16px',
-              },
-              background: i % 2 == 0 ? secondaryBackground : '#FAF8F3',
-              color: i % 2 == 0 ? secondaryColor : 'black',
-            }}
-          >
-            {child}
-          </Grid>
-        ))}
-      </Grid>
-    </div>
+            child={child}
+            verticalPadding={
+              i == 0 ? firstElemVerticalPadding : verticalPadding
+            }
+            color={i % 2 == 0 ? secondaryColor : theme.palette.text.primary}
+            background={
+              i % 2 == 0
+                ? secondaryBackground
+                : theme.palette.background.primary
+            }
```

Comment:
You repeat this ternary expression very often in this file. You can write a function to do that no ?

Reply:
Sure, I made the logic more explicit by creating two functions. See https://github.com/tournesol-app/tournesol/pull/1183/commits/5f37b36e3022f7e76b358ea9718073ab453d7448